capstone-rs/src/instruction.rs

```diff
@@ -328,6 +345,64 @@ impl<'a> InsnDetail<'a> {
     }
 }
 
+impl<'a> InsnRegsAccess {
+    fn new(cs: csh, ins: &cs_insn) -> Self {
+        let mut regs_read = [0u16; 64];
+        let mut regs_read_count = 0u8;
+        let mut regs_write = [0u16; 64];
+        let mut regs_write_count = 0u8;
+
+        unsafe {
+            cs_regs_access(
+                cs,
+                ins,
+                regs_read.as_mut_ptr(),
+                &mut regs_read_count,
+                regs_write.as_mut_ptr(),
+                &mut regs_write_count,
+            );
+        }
+
+        Self {
+            regs_read,
+            regs_read_count,
+            regs_write,
+            regs_write_count,
+        }
+    }
+
+    /// Returns the explicit and implicit accessed registers
+    pub fn regs_read(&self) -> RegsIter<RegIdInt> {
+        RegsIter(self.regs_read[..self.regs_read_count as usize].iter())
+    }
+
+    /// Returns the number of explicit and implicit read registers
+    pub fn regs_read_count(&self) -> u8 {
+        self.regs_read_count
+    }
+
+    /// Returns the explicit and implicit write registers
+    pub fn regs_write(&self) -> RegsIter<RegIdInt> {
+        RegsIter(self.regs_write[..self.regs_write_count as usize].iter())
+    }
+
+    /// Returns the number of explicit and implicit write registers
+    pub fn regs_write_count(&self) -> u8 {
+        self.regs_write_count
+    }
+}
+
+impl Debug for InsnRegsAccess {
```

Comment:
You can just derive the Debug impl; see above comment

Reply:
If you can't derive `Debug` because the arrays are too big (many of these impls are only defined for arrays up to a certain size), then add a comment explaining why you can't derive `Debug`.